hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -325,4 +340,111 @@ private void checkContainers(DatanodeDetails nodeByUuid, AtomicBoolean isContain
+      Preconditions.checkArgument(!ipAddress.isEmpty(),
+          "Either uuid or ipAddress of a datanode should be provided !!!");
+    }
+    try {
+      return getDecommissionStatusResponse(uuid, ipAddress);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Response getDecommissionStatusResponse(String uuid, String ipAddress) throws IOException {
+    Response.ResponseBuilder builder = Response.status(Response.Status.OK);
+    Map<String, Object> responseMap = new HashMap<>();
+    Stream<HddsProtos.Node> allNodes = scmClient.queryNode(DECOMMISSIONING,
+        null, HddsProtos.QueryScope.CLUSTER, "", ClientVersion.CURRENT_VERSION).stream();
+    List<HddsProtos.Node> decommissioningNodes = HddsUtils.getDecommissioningNodesList(allNodes, uuid, ipAddress);
+    String metricsJson = scmClient.getMetrics("Hadoop:service=StorageContainerManager,name=NodeDecommissionMetrics");
+    int numDecomNodes = -1;
+    JsonNode jsonNode = null;
+    if (metricsJson != null) {
+      jsonNode = HddsUtils.getBeansJsonNode(metricsJson);
+      numDecomNodes = HddsUtils.getNumDecomNodes(jsonNode);
+    }
+    List<Map<String, Object>> dnDecommissionInfo =
+        getDecommissioningNodesDetails(decommissioningNodes, jsonNode, numDecomNodes);
+    try {
+      responseMap.put("DatanodesDecommissionInfo", dnDecommissionInfo);
+      builder.entity(responseMap);
+      return builder.build();
+    } catch (Exception exception) {
+      LOG.error("Unexpected Error: {}", exception);
+      throw new WebApplicationException(exception, Response.Status.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private List<Map<String, Object>> getDecommissioningNodesDetails(List<HddsProtos.Node> decommissioningNodes,
+                                                                   JsonNode jsonNode,
+                                                                   int numDecomNodes) throws IOException {
+    List<Map<String, Object>> decommissioningNodesDetails = new ArrayList<>();
+
+    for (HddsProtos.Node node : decommissioningNodes) {
+      DatanodeDetails datanode = DatanodeDetails.getFromProtoBuf(
+          node.getNodeID());
+      Map<String, Object> datanodeMap = new LinkedHashMap<>();
+      datanodeMap.put("datanodeDetails", datanode);
+      datanodeMap.put("metrics", getCounts(datanode, jsonNode, numDecomNodes));
+      datanodeMap.put("containers", getContainers(datanode));
+      decommissioningNodesDetails.add(datanodeMap);
+    }
+    return decommissioningNodesDetails;
+  }
+
+  private Map<String, Object> getCounts(DatanodeDetails datanode, JsonNode counts, int numDecomNodes) {
+    Map<String, Object> countsMap = new LinkedHashMap<>();
+    String errMsg = getErrorMessage() + datanode.getHostName();
+    try {
+      countsMap = HddsUtils.getCountsMap(datanode, counts, numDecomNodes, countsMap, errMsg);
+      if (countsMap != null) {
+        return countsMap;
+      }
+      LOG.error(errMsg);
```

Comment:
is there any error message for success case?

Reply:
> is there any error message for success case?
No,  this is not success case. For success case. it is being returned from Line 427 itself.